packages/blade/src/components/Box/BaseBox/getBaseBoxStyles.ts

```diff
@@ -1,102 +1,260 @@
+const shouldAddBreakpoint = (cssProps: CSSObject): boolean => {
+  const firstDefinedValue = Object.values(cssProps).find(
+    (cssValue) => cssValue !== undefined && cssValue !== null,
+  );
+
+  return firstDefinedValue !== undefined;
+};
+
+const getAllMediaQueries = (props: BaseBoxProps & { theme: Theme }): CSSObject => {
+  if (isReactNative()) {
+    return {};
+  }
+
+  const { base, ...breakpointsWithoutBase } = breakpoints;
+
+  return Object.fromEntries(
+    Object.entries(breakpointsWithoutBase).map(([breakpointKey, breakpointValue]) => {
+      const cssPropsForCurrentBreakpoint = getAllProps(props, breakpointKey as keyof Breakpoints);
+      if (!shouldAddBreakpoint(cssPropsForCurrentBreakpoint)) {
+        return [];
+      }
+
+      const mediaQuery = `@media ${getMediaQuery({ min: breakpointValue })}`;
+      return [mediaQuery, cssPropsForCurrentBreakpoint];
+    }),
+  );
+};
+
+const getBaseBoxStyles = (props: BaseBoxProps & { theme: Theme }): CSSObject => {
+  return {
+    ...getAllProps(props),
+    ...getAllMediaQueries(props),
+  };
 };
 
-const getBaseBoxStyles = ({
-  theme,
-  display,
-  flex,
-  flexWrap,
-  flexDirection,
-  flexGrow,
-  flexShrink,
-  alignItems,
-  alignContent,
-  justifyContent,
-  alignSelf,
-  overflow,
-  paddingTop,
-  paddingBottom,
-  paddingLeft,
-  paddingRight,
-  marginTop,
-  marginBottom,
-  marginLeft,
-  marginRight,
-  gap,
-  width,
-  height,
-  minHeight,
-  minWidth,
-  maxHeight,
-  maxWidth,
-  position,
-  transform,
-  top,
-  left,
-  right,
-  bottom,
-  background,
-  backgroundColor,
-  zIndex,
-  borderRadius,
-}: BaseBoxProps & { theme: Theme }): CSSObject => ({
-  display,
-  flex,
-  flexWrap,
-  flexGrow,
-  flexShrink,
-  flexDirection,
-  alignItems,
-  alignContent,
-  justifyContent,
-  alignSelf,
-  overflow,
-  position,
-  top,
-  right,
-  bottom,
-  left,
-  paddingTop: getSpacingValue({ value: paddingTop, theme }),
-  paddingBottom: getSpacingValue({ value: paddingBottom, theme }),
-  paddingLeft: getSpacingValue({ value: paddingLeft, theme }),
-  paddingRight: getSpacingValue({ value: paddingRight, theme }),
-  marginTop: getSpacingValue({ value: marginTop, theme }),
-  marginBottom: getSpacingValue({ value: marginBottom, theme }),
-  marginLeft: getSpacingValue({ value: marginLeft, theme }),
-  marginRight: getSpacingValue({ value: marginRight, theme }),
-  gap: getSpacingValue({ value: gap, theme }),
-  width,
-  height,
-  minHeight: minHeight ? makeSize(minHeight) : undefined,
-  minWidth: minWidth ? makeSize(minWidth) : undefined,
-  maxHeight: maxHeight ? makeSize(maxHeight) : undefined,
-  maxWidth: maxWidth ? makeSize(maxWidth) : undefined,
-  transform,
-  background,
-  backgroundColor,
-  zIndex,
-  borderRadius: makeBorderSize(getIn(theme, `border.radius.${borderRadius}`)),
-});
-
-export default getBaseBoxStyles;
+const getDependencyProps = (props: BaseBoxProps & { theme: Theme }): string | BaseBoxProps => {
```

Comment:
Can you add a comment here describing what this function does? Would be helpful for the future

Reply:
we moved this to new utility in styled-props PR. Have added comments there - https://github.com/razorpay/blade/blob/bd61a389316043bce1e09610c90ef16734867795/packages/blade/src/components/Box/BaseBox/useMemoizedStyles.web.ts#L7-L47